support_util/support-utils.sh

```diff
@@ -0,0 +1,375 @@
+        $k8sCmd -n "${namespace}" logs "${pod}" -c "${SYSDIG_HS_CONTAINER_NAME}" > "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${pod}_${SYSDIG_HS_CONTAINER_NAME}.log" && \
+        log_activity "Collected host-scanner logs for pod ${pod}." || \
+        log_activity "host-scanner not installed for pod ${pod}."
+        $k8sCmd -n "${namespace}" logs "${pod}" -c "${SYSDIG_RS_CONTAINER_NAME}" > /dev/null 2>&1 && \
+        $k8sCmd -n "${namespace}" logs "${pod}" -c "${SYSDIG_RS_CONTAINER_NAME}" > "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${pod}_${SYSDIG_RS_CONTAINER_NAME}.log" && \
+        log_activity "Collected runtime-scanner logs for pod ${pod}." || \
+        log_activity "runtime-scanner not installed for pod ${pod}."
+      fi
+      if [[ $pod =~ $SYSDIG_KSPMC_PREFIX ]]; then
+        mkdir -p "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs"
+        $k8sCmd -n "${namespace}" logs "${pod}" > "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${pod}.log" 2>/dev/null
+        if [ $? -eq 0 ]; then
+          log_activity "Collected kspmcollector log for pod ${pod}."
+        else
+          log_activity "ERROR: Failed to collect kspmcollector log for pod ${pod} (pod not found)."
+        fi
+      fi
+    done
+    log_activity "Log collection completed for all pods."
+  else
+    log_activity "Attempting log collection for pod ${podName}."
+    if [[ $podName != *"clustershield"* && $podName != *"node-analyzer"* && $podName != *"shield-cluster"* && $podName != *"kspmcollector"* && $podName =~ $SYSDIG_AGENT_PREFIX ]]; then
+      printf "Collecting log for pod ${podName}\n"
+      mkdir -p "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${podName}"
+      $k8sCmd -n "${namespace}" cp "${podName}:${AGENT_LOG_DIR}." "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${podName}" 2>/dev/null
+      if [ $? -eq 0 ]; then
+        printf "Log collected for pod ${podName}\n"
+        log_activity "Collected agent log for pod ${podName}."
+      else
+        log_activity "ERROR: Failed to collect agent log for pod ${podName} (possibly recycled)."
+      fi
+    fi
+    if [[ $podName =~ $SYSDIG_CS_PREFIX || $podName =~ $SYSDIG_CS_SH_PREFIX ]]; then
+      printf "Collecting log for pod ${podName}\n"
+      mkdir -p "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${SYSDIG_CS_DIR}"
+      $k8sCmd -n "${namespace}" logs "${podName}" > "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs/${SYSDIG_CS_DIR}/${podName}.log" 2>/dev/null
+      if [ $? -eq 0 ]; then
+        printf "Log collected for pod ${podName}\n"
+        log_activity "Collected cluster shield log for pod ${podName}."
+      else
+        log_activity "ERROR: Failed to collect cluster shield log for pod ${podName} (pod not found)."
+      fi
+    fi
+    if [[ $podName =~ $SYSDIG_NA_PREFIX ]]; then
+      printf "Collecting log for pod ${podName}\n"
+      mkdir -p "${DEST_DIR}/${SYSDIG_SUPPORT_DIR}/logs"
+      $k8sCmd -n "${namespace}" logs "${podName}" -c "${SYSDIG_KSPMA_CONTAINER_NAME}" > /dev/null 2>&1 && \
```

Comment:
You should never need quotes for namespace or podName or containerName. Also please be consistent using variables. Always either `${var}` or `$var`